### ISSUE TITLE
Logs interface hangs with long log output

### ISSUE THREAD

**Issue by glmdev** [ohwgiles/laminar#165]
Hello! Thanks for such a polished & lightweight tool.

I've been using Laminar for automated Angular builds, triggered via git hooks. With larger projects and newer versions of Angular, the verbose output can become quite long.

When opening such long output in the web interface in Laminar, RAM usage spikes dramatically and the tab hangs until closed.

This isn't a particularly big issue as I can redirect the build output to a file in the `$ARCHIVE` directory to download and view it in a text editor instead, but I wonder if there is some way to make the front-end handle this case more gracefully? This is kind of an edge case, so might not be worth it.

Edit: using Laminar 1.1 on CentOS 7

**Comment by ohwgiles** [comment]
Thanks for your feedback.

About how big are these log files?

When you say RAM usage spikes dramatically, do you mean the memory of the browser or of `laminard`?

Is the behaviour any different between visiting a completed run log vs watching one in progress?

**Comment by glmdev** [comment]
> About how big are these log files?

I've noticed the issue with log files of 6-8 MiB and larger. For the case that prompted this report, the `output` column in the `builds` table is ~10 MiB.

> When you say RAM usage spikes dramatically, do you mean the memory of the browser or of `laminard`?

By this, I mean that the memory usage of the browser spikes dramatically, not `laminard`. By the time I closed the tab, it was using some 10% of system memory, which would be a bit under a GiB.

> Is the behaviour any different between visiting a completed run log vs watching one in progress?

The case I encountered was for a completed log. For in-progress logs, it will work fine for a while, then the loading of additional log chunks will become choppy, and eventually the log section of the screen will disappear and the log will stop rendering.
